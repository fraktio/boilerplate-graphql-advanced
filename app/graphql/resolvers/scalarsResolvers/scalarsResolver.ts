import {
  EmailAddressResolver,
  PhoneNumberResolver,
  UUIDResolver,
} from "graphql-scalars";

import { Resolvers } from "~/generated/graphql";
import { DateResolver } from "~/graphql/resolvers/scalarsResolvers/DateResolver";
import { DateTimeResolver } from "~/graphql/resolvers/scalarsResolvers/DateTimeResolver";

export const scalarsResolver: Resolvers = {
  UUID: UUIDResolver,
  Date: DateResolver,
  DateTime: DateTimeResolver,
  PhoneNumber: PhoneNumberResolver,
  EmailAddress: EmailAddressResolver,
};
