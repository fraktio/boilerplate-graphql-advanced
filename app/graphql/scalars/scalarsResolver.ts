import { UUIDResolver } from "graphql-scalars";

import { CountryCodeResolver } from "./CountryCodeResolver";
import { DateResolver } from "./DateResolver";
import { DateTimeResolver } from "./DateTimeResolver";
import { EmailResolver } from "./EmailResolver";
import { PhoneNumberResolver } from "./PhoneNumberResolver";

import { Resolvers } from "~/generation/generated";

export const createScalarResolvers = (): Resolvers => ({
  UUID: UUIDResolver,
  Date: DateResolver,
  DateTime: DateTimeResolver,
  EmailAddress: EmailResolver,
  PhoneNumber: PhoneNumberResolver,
  CountryCode: CountryCodeResolver,
});
