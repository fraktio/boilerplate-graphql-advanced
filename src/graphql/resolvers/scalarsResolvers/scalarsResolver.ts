import { EmailAddressResolver, PhoneNumberResolver } from "graphql-scalars";

import { Resolvers } from "~/generated/graphql";
import { DateResolver } from "~/graphql/resolvers/scalarsResolvers/DateResolver";
import { DateTimeResolver } from "~/graphql/resolvers/scalarsResolvers/DateTimeResolver";

export const scalarsResolver: Resolvers = {
  Date: DateResolver,
  DateTime: DateTimeResolver,
  PhoneNumber: PhoneNumberResolver,
  EmailAddress: EmailAddressResolver,
};
