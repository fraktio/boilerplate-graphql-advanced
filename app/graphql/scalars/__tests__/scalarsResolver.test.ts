import { createScalarResolvers } from "../scalarsResolver";

describe("graphql scalars / scalarsResolver", () => {
  it("createScalarResolvers", async () => {
    expect(createScalarResolvers()).toMatchSnapshot();
  });
});
