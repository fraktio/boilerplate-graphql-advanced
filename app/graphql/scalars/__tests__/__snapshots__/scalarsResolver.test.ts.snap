// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql scalars / scalarsResolver createScalarResolvers 1`] = `
Object {
  "Date": "Date",
  "DateTime": "DateTime",
  "EmailAddress": "Email",
  "PhoneNumber": "Phone",
  "UUID": "UUID",
}
`;
